gdnative-core/src/private.rs

```diff
@@ -1,9 +1,15 @@
+use std::any::TypeId;
+use std::collections::HashSet;
 use std::ffi::CString;
 
+use crate::nativescript::NativeClass;
 use crate::sys;
+use once_cell::sync::Lazy;
+use parking_lot::RwLock;
 
 static mut GODOT_API: Option<sys::GodotApi> = None;
 static mut GDNATIVE_LIBRARY_SYS: Option<*mut sys::godot_object> = None;
+static CLASS_REGISTRY: Lazy<RwLock<HashSet<TypeId>>> = Lazy::new(|| RwLock::new(HashSet::new()));
```

Comment:
Is it necessary to put this in `gdnative::private`, which is a place for private interfaces that *need* to be technically `pub`? I don't see this used outside `gdnative-core`, so it can be placed in a regular private module instead, I think.

Reply:
Oh, I see. I put it in `gdnative::private` because I was under the impression that it was related to the internal implementation of `gdnative-core` and it looked like a good place to put it directly with the API.

Would it be fine to move the references into `gdnative-core::init::{new_private_module}`?

Reply:
One other note: I put it in the private.rs because I thought that the registry would be considered "internal state" and need to be cleaned up via `gdnative::private::cleanup_internal_state()`. I'd be happy to change that and add in a cleanup function that can be called from that state though.

Reply:
@toasteater should I go ahead and proceed with making those changes?

Reply:
@jacobsky `gdnative_core::native_script::class_registry` should be fine. `gdnative_core::native_script::registry` should be fine too, at least until we need more registries. This is purely internal, so naming is not all that important -- we can change that at any time.

> the registry would be considered "internal state" and need to be cleaned up

Yes, although you may also do that for a private module, like what's done for the type tags: `crate::nativescript::type_tag::cleanup()`.

This also reminded me that you might want to keep the call behind a `#[cfg(feature = "nativescript")]`, since it should be possible to compile the library without NativeScript support.

Reply:
Thank you for all of your insight! I just finished up working on those changes. Please let me know if you see any other issues.